core/src/main/java/com/google/cloud/sql/core/InternalConnectorRegistry.java

```diff
@@ -261,4 +267,55 @@ private Connector createConnector(ConnectionConfig config) {
         refreshTimeoutMs,
         serverProxyPort);
   }
+
+  private ConnectionConfig resolveConfig(ConnectionConfig config) {
+    if (config.getNamedConnection() != null) {
+      ConnectorKey namedConfig = resolveKey(config);
+      if (namedConfig == null) {
+        throw new IllegalArgumentException(
+            "Named configuration " + config.getNamedConnection() + " does not exist.");
+      }
+      return namedConfig.getConfig();
+    }
+    return config;
+  }
+
+  private ConnectorKey resolveKey(ConnectionConfig config) {
+    Optional<ConnectorKey> namedConfig =
+        connectors.keySet().stream()
+            .filter(k -> config.getNamedConnection().equals(k.getConfig().getNamedConnection()))
+            .findFirst();
+    return namedConfig.orElse(null);
+  }
+
+  /** Register the configuration for a named connector. */
+  public void register(String name, ConnectionConfig config) {
+    ConnectionConfig resolvedConfig = config.withNamedConnection(name);
+    ConnectorKey key = new ConnectorKey(resolvedConfig);
+    if (this.connectors.containsKey(key)) {
```

Comment:
To verify if this name is already being used we should not consider the entire ConnectionConfig. 

The way it is implemented, the following will not throw an exception:

```
ConnectionConfig config =  new ConnectionConfig.Builder()
            .withCloudSqlInstance("myProject:myRegion:myInstance")
            .withIpTypes("PRIMARY")
            .build();
ConnectorRegistry.getConnector().register("my-connection", config);

ConnectionConfig config2 =  new ConnectionConfig.Builder()
            .withCloudSqlInstance("myProject:myRegion:myInstance")
            .withIpTypes("PRIVATE")
            .build();
ConnectorRegistry.getConnector().register("my-connection", config2);
```

Reply:
Great catch. This is a bug. I didn't see this when I rewrote the connector map behavior. 

To get the correct behavior the InternalConnectorRegistry will need to store the ConnectionConfig and Connector by name. This is why I originally implemented this with two maps: a map of Connector with "name" as a key, and a map of ConnectionConfig with name as a key. 

Maybe we need to revisit whether "one map for all connectors" will get us the behavior we want.

Reply:
I have added a new test: `InternalConnectorRegistryTest.registerConnectionFailsWithDuplicateNameAndDifferentConfig()`

Reply:
I updated the implementation of `InternalConnectorRegistry` so that it has two maps again and the new test will now pass. Let me know what you think.